Src/IronPython/Runtime/CodeContext.cs

```diff
@@ -20,15 +20,17 @@ namespace IronPython.Runtime {
     public sealed class CodeContext {
         private readonly ModuleContext/*!*/ _modContext;
         private readonly PythonDictionary/*!*/ _dict;
+        private readonly CodeContext _maybegrandparent;// can be null.
 
         /// <summary>
         /// Creates a new CodeContext which is backed by the specified Python dictionary.
         /// </summary>
-        public CodeContext(PythonDictionary/*!*/ dict, ModuleContext/*!*/ moduleContext) {
+        public CodeContext(PythonDictionary/*!*/ dict, ModuleContext/*!*/ moduleContext, CodeContext maybegrandparent) {
```

Comment:
Is this a concept that should be exposed publicly? Maybe add an internal constructor overload instead?